src/tsc/tsc.ts

```diff
@@ -394,6 +396,17 @@ namespace ts {
         }
     }
 
+    function makeFilesReferencesAbsolute(optionsFromCLI: CompilerOptions) {
+        Object.keys(optionsFromCLI).forEach(key => {
+            const optionForKey = getOptionFromName(key, /*allowShort*/ true);
+            const value = optionsFromCLI[key];
+            const relative = isString(value) && !isRootedDiskPath(value);

```

Comment:
I think you should rather use `convertOptionsFromJson` instead of doing the conversion here (errors ignored).
Make that function internal so you can add tests for correct result.
Sorry I didn't check if such function exists earlier.

Reply:
👍  sure thing, thanks for the fast feedback!